src/hotspot/os/windows/os_windows.cpp

```diff
@@ -5315,6 +5315,28 @@ void os::funlockfile(FILE* fp) {
   _unlock_file(fp);
 }
 
+char* os::realpath(const char* filename, char* outbuf, size_t outbuflen) {
+
+  if (filename == nullptr || outbuf == nullptr || outbuflen < 1) {
+    assert(false, "os::realpath: invalid arguments.");
+    errno = EINVAL;
+    return nullptr;
+  }
+
+  char* result = nullptr;
+  ALLOW_C_FUNCTION(::_fullpath, char* p = ::_fullpath(nullptr, filename, 0);)
+  if (p != nullptr) {
+    if (strlen(p) < outbuflen) {
+      strcpy(outbuf, p);
+      result = outbuf;
+    } else {
+      errno = ENAMETOOLONG;
+    }
```

Comment:
Curious, why not just passing in outbuf and outbuflen directly instead of letting the function allocate memory just to then copy the content? To get a guaranteed ENAMETOOLONG?

Just a question, not a comment. I am fine with this as it is.

Reply:
That's exactly it - I don't know what different versions of Windows do.  I do know, for instance, that my Windows 10 vm sets errno to ERANGE (I could simplify the code and just test and convert that, but I'm being cautious, I guess).